compiler-core/src/ast/inlay_hints.rs

```diff
@@ -0,0 +1,104 @@
+use crate::{
```

Comment:
The main logic is implemented here instead of `language_server::*` in order to be tested with unit tests instead of integration tests. Integration snapshots tests with LSP are included too (but there are fewer of them)

Reply:
This is part of the language server, not part of the core compiler, so it must be in the language server part of the codebase.